From 9ccdf5c095adb6eb60b7e96bd1e8bac342b5d685 Mon Sep 17 00:00:00 2001
From: Omar Ramirez Luna <omar.r.ramirez.luna@intel.com>
Date: Tue, 25 Oct 2016 23:29:33 -0700
Subject: [PATCH 23/27] ov8858: fix regulator management on ov8858_s_ctrl

ov8858_s_ctrl (called from v4l2_ctrl_handler_setup during probe)
communicates through i2c with the sensor. At thid point the voltages
have been powered off and this causes a timeout in the i2c line.

This will fix the bug uncovered by the usage of the regulator
framework:

ov8858 i2c-INT3477:00: gmin: initializing atomisp module subdev data.PMIC ID 1
ov8858 i2c-INT3477:00: camera pdata: port: 1 lanes: 4 order: 00000002
i2c_designware 808622C1:03: controller timed out
ov8858 i2c-INT3477:00: write DW9718_DATA_M  failed -5

Signed-off-by: Omar Ramirez Luna <omar.r.ramirez.luna@intel.com>
---
 drivers/media/i2c/ov8858.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/media/i2c/ov8858.c b/drivers/media/i2c/ov8858.c
index 8001995..a1fc14d 100644
--- a/drivers/media/i2c/ov8858.c
+++ b/drivers/media/i2c/ov8858.c
@@ -2263,8 +2263,17 @@ static int ov8858_probe(struct i2c_client *client,
 	/* Use same lock for controls as for everything else. */
 	dev->ctrl_handler.lock = &dev->input_lock;
 	dev->sd.ctrl_handler = &dev->ctrl_handler;
+
+	ret = __ov8858_s_power(&dev->sd, 1);
+	if (ret) {
+		dev_err(&client->dev, "power-up error %d!\n", ret);
+		return ret;
+	}
+
 	v4l2_ctrl_handler_setup(&dev->ctrl_handler);
 
+	__ov8858_s_power(&dev->sd, 0);
+
 	ret = media_entity_init(&dev->sd.entity, 1, &dev->pad, 0);
 	if (ret) {
 		ov8858_remove(client);
-- 
2.9.3

